src/span.h

```diff
@@ -23,7 +23,12 @@ class Span
     constexpr Span(C* data, std::ptrdiff_t size) noexcept : m_data(data), m_size(size) {}
 
     constexpr C* data() const noexcept { return m_data; }
+    constexpr C* begin() const noexcept { return m_data; }
+    constexpr C* end() const noexcept { return m_data + m_size; }
     constexpr std::ptrdiff_t size() const noexcept { return m_size; }
+    constexpr C& operator[](std::ptrdiff_t pos) const noexcept { return m_data[pos]; }
```

Comment:
Just a thought: wouldn't asserting `pos < m_size` be a quick win here to catch out of bounds bugs during development?

Reply:
Sounds like a good idea to add in a form of debug mode (like `DEBUG_ADDRMAN` etc), but perhaps independently of this PR?

Reply:
Yes, we already have that; see DEBUG_ADDRMAN, DEBUG_LOCKORDER, DEBUG_LOCKCONTENTION. More could be added.